libraries/core_libs/network/src/tarcap/packets_handlers/common/ext_votes_packet_handler.cpp

```diff
@@ -53,6 +59,12 @@ void ExtVotesPacketHandler::sendPbftNextVotes(dev::p2p::NodeID const &peer_id,
       }
     }
   }
+
+  if (send_next_votes_bundle.size() - index > kMaxVotesInPacket) {
+    periodic_events_tp_->post(100, [this, index, peer_id, send_next_votes_bundle] {
```

Comment:
So we have cases where we might be sending very large number of votes or transactions. Usually this only happens after a restart or if a network got stuck for some time. Any thread that sends it in one or more packets would be blocked for some time. Currently we have this one thread in periodic_events_tp_ thread pool. So I think we either need to have multiple threads so that this kind of long operation does not block anything else or we have less threads but delay between sending so that other stuff can be done at the same time as well. Also there is always consideration about the receiving end. If we are sending more transactions/votes than what the other side can process there will be some packets buffering on the other end but maybe we are fine with that.

Reply:
I have removed delays from sending multiple packets